### ISSUE TITLE
Bearer middleware return 200 when IDP not reachable.

### ISSUE THREAD

**Issue by karoldeland** [dapr/components-contrib#2599]
## Expected Behavior

When IDP url is unreachable, I expect to have a 401 ou other to say that request can't be validated.

## Actual Behavior

Receiving 200 OK.

## Steps to Reproduce the Problem

Follow this readme but skip launching IdentityServer.
https://github.com/karoldeland/daprShowcase/tree/main/ServiceInvocation

Tested with 1.10.2

## Release Note

RELEASE NOTE:

**Comment by ItalyPaleAle** [comment]
If the IDP is completely unreachable (e.g. URL doesn't resolve to anything), the component fails to initialize. You should see in the logs something like:

```
WARN[0000] failed to build HTTP pipeline: error creating HTTP middleware middleware.http.bearer/: failed to fetch JWKS: failed to fetch "https://google.com": failed to transform HTTP response for "https://google.com": failed to parse JWK set at "https://google.com": failed to unmarshal JWK set: invalid character '<' looking for beginning of value  app_id=middleware instance=AleTestVM scope=dapr.runtime type=log ver=dev
```

The reason why you get 200's is because Dapr just ignores the middleware that failed and continues the initialization regardless.

So, to me there are 2 sides to this issue:

- First is a component-level issue. Considering that failures may be transient, we should not fail to init the component if the IDP doesn't resolve, but like you suggested make it work and return 401s
- The second is a runtime-level issue. Perhaps Dapr shouldn't complete the initialization if a middleware fails to init.

**Comment by ItalyPaleAle** [comment]
The fix for this is two-fold.

First, the runtime issue is being fixed by: https://github.com/dapr/dapr/pull/5993
Once this  is merged, if a middleware component fails to init, Dapr won't start.

Second, there's an issue specifically with the bearer middleware, which in Dapr 1.10 does not check if the IDP is valid upon initialization. This is actually already fixed in master by #2538

**Comment by ItalyPaleAle** [comment]
Fixed in Dapr 1.10.3
